scripts/public_api_report.py

```diff
@@ -784,6 +854,394 @@ class CodeReferenceFilter:
+            name="head",
+            filepath=pathlib.Path("great_expectations/validator/metrics_calculator.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for Python `Set.add()`",
+            name="add",
+            filepath=pathlib.Path("great_expectations/validator/validation_graph.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for Python dict `.update()`",
+            name="update",
+            filepath=pathlib.Path("great_expectations/validator/validation_graph.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path(
+                "great_expectations/checkpoint/types/checkpoint_result.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path("great_expectations/core/batch.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path("great_expectations/core/domain.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path(
+                "great_expectations/core/expectation_configuration.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path(
+                "great_expectations/core/expectation_diagnostics/expectation_diagnostics.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path("great_expectations/core/expectation_suite.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path(
+                "great_expectations/core/expectation_validation_result.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Internal use",
+            name="IDDict",
+            filepath=pathlib.Path("great_expectations/core/id_dict.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path("great_expectations/core/run_identifier.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="to_json_dict is an internal helper method",
+            name="to_json_dict",
+            filepath=pathlib.Path("great_expectations/data_context/types/base.py"),
+        ),
```

Comment:
I think these should be public. They are used for passing things around in the Airflow Operator, and some users use it when they need to pass around various forms of config. My hope is that the need for it goes away when we transition to New Style Datasources, but we’re not quite there yet.